### PR TITLE
HAWKULAR-982 - Added a bunch of logs to most of the operations.

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

**Comment by jpkrohling** [comment]
I tried to follow these rules:
- Log every operation that changes state (create, update, delete)
- Log moderate to complex logic steps (get a list of things then subtract from list of things)
- Log error messages from REST endpoints
- Do not log error messages before exceptions are thrown. This is because the exception will already be seen on the logs, so, no point in having a log right before that.
- Log as much as possible backend HTTP interactions
- Log as much as possible operations that occurs asynchronously (or in the background)

All new log entries are in TRACE mode. If any message proves to be helpful in a higher level, we change the level later.

File: common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java [hunk 1]
```diff
@@ -63,10 +67,12 @@ public String execute(String url, String clientId, String secret, String method)
      */
     public String execute(String url, String urlParameters, String clientId, String secret, String method) throws
             Exception {
+        logger.executingAuthServerRequest(url, urlParameters, clientId, secret, method);
 
         HttpURLConnection connection;
         String credentials = clientId + ":" + secret;
         String authorizationHeader = "Basic " + Base64.getEncoder().encodeToString(credentials.getBytes());
+        logger.authHeaderConvertedTo(clientId, secret, authorizationHeader);
```
Comment (jpkrohling) [inline common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java]: Should we really log this?

File: common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java [hunk 2]
```diff
@@ -63,10 +67,12 @@ public String execute(String url, String clientId, String secret, String method)
      */
     public String execute(String url, String urlParameters, String clientId, String secret, String method) throws
             Exception {
+        logger.executingAuthServerRequest(url, urlParameters, clientId, secret, method);
```
Comment (jpkrohling) [inline common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java]: Should we really log the secret?

Comment (jpkrohling) [inline common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java]: Also, the URL Parameters might include an user's password. Should we log it?

Comment (pilhuhn) [inline common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java]: no

File: common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java (hunk 1 above)
Comment (pilhuhn) [inline common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java]: Certainly not the secret.

**Comment by pilhuhn** [comment]
Did you check that with jboss-logging no guards like

```
if (log.isEnabled(Logger.Level.TRACE))
```

are needed? Otherwise Logging may end up being a huge performance hit.

**Comment by jpkrohling** [comment]
No guards like that are needed, as they are done in the actual implementation:

```
    if(this.isEnabled(level)) {
        this.doLogf(level, loggerFqcn, format, new Object[]{param1, param2}, t);
    }
```

File: common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java (hunk 1 above)
Comment (jpkrohling) [inline common/src/main/java/org/hawkular/accounts/common/AuthServerRequestExecutor.java]: As I'm logging the encoded auth header (which can be easily decoded back), I'll just not log it then.

**Pull request merged**